src/core/mac/mac.cpp

```diff
@@ -275,11 +278,31 @@ void Mac::PerformActiveScan(void)
     {
         mSubMac.SetPanId(mPanId);
         FinishOperation();
-        mActiveScanHandler(GetInstance(), NULL);
+        ReportActiveScanResult(NULL);
         PerformNextOperation();
     }
 }
 
+void Mac::ReportActiveScanResult(RxFrame *aBeaconFrame)
```

Comment:
Is it possible to
```suggestion
void Mac::ReportActiveScanResult(const RxFrame *aBeaconFrame)
```

Reply:
👍  was thinking of the same. 
It requires adding some changes in `Beacon` class and `ConvertBeaconToActiveScanResult` (missing  some `const` defs/method). Will add it later as a separate commit.

Reply:
Added it in the main commit itself.